components/eventing-controller/controllers/subscription/jetstream/reconciler.go

```diff
@@ -248,6 +248,10 @@ func (r *Reconciler) syncInitialStatus(subscription *eventingv1alpha1.Subscripti
 	cleanedSubjects, err := handlers.GetCleanSubjects(subscription, r.eventTypeCleaner)
 	if err != nil {
 		log.Errorw("getting clean subjects failed", "error", err)
+		if len(subscription.Status.CleanEventTypes) != 0 {
+			subscription.Status.CleanEventTypes = nil
```

Comment:
Can we make the field CleanEventTypes to be not `omitEmpty` ?
https://github.com/kyma-project/kyma/blob/9046447c9818993c2107b3c81d3fcb3276d1fe64/components/eventing-controller/api/v1alpha1/subscription_types.go#L189

My argumentation for this would be that it is easier to spot that a value is empty and hence the Subscription is not ready in this aspect, instead of checking if the field is not existing in the Subscription. Like the motto: "What I don't see is out of my mind" :-D

Reply:
I agree. But the minute we do it, we have failing tests for reasons unknown. I will create a followup issue for it as discussed.

Reply:
https://github.com/kyma-project/kyma/issues/14293